src/integrated/land.jl

```diff
@@ -226,6 +239,8 @@ the components returned by `land_components!`.
 This needs to be fixed.
 """
 ClimaLand.land_components(land::LandModel) = (:soil, :snow, :soilco2, :canopy)
+ClimaLand.land_components(land::LandModel{FT, Nothing}) where {FT} =
```

Comment:
maybe it would be better to have this be like
```
function ClimaLand.land_components(land::LandModel)
    land.soil_co2 isa Nothing ? (:soil, :snow, :canopy) : (:soil, :snow, :soilco2, :canopy)
end
```

like Julia suggested... is this slightly more readable? Up to you. Maybe it doesnt take advantage of the types in the same way